src/scripts/turnmarker.js

```diff
@@ -111,14 +113,50 @@ Hooks.on('updateToken', async (scene, updateToken, updateData) => {
     }
 });
 
-Hooks.on('updateTile', async () => {
-    if (canvas.scene.data.tokenVision) {
-        let tile = canvas.tiles.placeables.find(t => t.data.flags.turnMarker == true);
+function isVisible(tile) {
+    if (tile.data.hidden) {
+        return game.user.isGM;
+    }
+
+    if (!canvas.sight.tokenVision) {
+        return true;
+    }
+
+    if (tile._controlled) {
+        return true;
+    }
+
+    const combatant = canvas.tokens.placeables.find(t => t.id === game.combat.combatant.tokenId);
+
+    if (combatant?.data.hidden) {
+        return game.user.isGM;
+    }
+
+    if (combatant._controlled) {
+        return true;
+    }
+
+    const ratio = Settings.getRatio();
+    const w = tile.data.width / ratio;
+    const h = tile.data.height / ratio;
+    const tolerance = Math.min(w, h) / 4;
+
+    return canvas.sight.testVisibility(tile.center, { tolerance, object: tile });
+}
+
+Hooks.on('updateTile', (entity, data, options, userId) => {
```

Comment:
Can we drop async here? I was pretty sure this was part of the culprit in fixing:

https://github.com/kckaiwei/TurnMarker-alt/issues/13
(Need to confirm it doesn't break again though)
I am however still fairly new to Foundry modding, so feel free to explain if I'm wrong.

Reply:
Nevermind, it seems that my addition of async previously wasn't actually necessary to the fixing of that previous bug.